packages/graphql_server/src/scraping/twitterScraping.ts

```diff
@@ -0,0 +1,70 @@
+import axios from 'axios'
+import { TwitterPost, TwitterSearchResponse } from '../../types/twitterScraping'
+
+const apiKey = process.env.SCRAPER_API_API_KEY || ''
+const SORT_BY_RETWEETS = 'retweets'
+const SORT_BY_REPLIES = 'replies'
+
+/**
+ * Gets first page of twitter posts for a given hashtag
+ * @param hashtag - the hashtag to search for
+ * @returns An array of the TwitterPosts
+ */
+export async function getPostsForHashtag(hashtag: string): Promise<TwitterPost[]> {
+  const { data } = await axios.get<TwitterSearchResponse>(
+    'https://api.scraperapi.com/structured/twitter/v2/search',
+    {
+      params: {
+        api_key: apiKey,
+        query: hashtag
+      }
+    }
+  )
+  return mapToInternalType(data.tweets)
+}
+
+/**
+ * Gets first page of twitter posts for a given hashtag and sorts the tweets by given criterion
+ * @param hashtag - the hashtag to search for
+ * @param sortBy - the sorting criterion
+ * @returns An sorted array of the TwitterPosts
+ */
+export async function getPostsForHashtagSortedBy(
+  hashtag: string,
+  sortBy: string
+): Promise<TwitterPost[]> {
+  const tweets = await getPostsForHashtag(hashtag)
+
+  return tweets.sort((a, b) => {
+    if (sortBy === SORT_BY_REPLIES) {
+      return b.replies - a.replies
+    }
+    if (sortBy === SORT_BY_RETWEETS) {
+      return b.retweetCount - a.retweetCount
+    }
+    return 0
+  })
+}
+
+/**
+ * Maps an array of the API response from TwitterSearchResponse type to TwitterPost type.
+ * @param tweets - The array of tweets to be mapped.
+ * @returns An array of mapped TwitterPost objects.
+ */
+function mapToInternalType(tweets: TwitterSearchResponse['tweets']): TwitterPost[] {
+  return tweets.map((tweet) => ({
+    id: tweet.tweet_id ?? '',
+    userId: tweet.user_id ?? '',
+    userName: tweet.user_name ?? '',
+    realName: tweet.real_name ?? '',
+    userVerified: tweet.user_verified ?? false,
+    conversationId: tweet.conversation_id ?? '',
+    // the URL given from the API call is wrongly formatted, so we have to build the url to the post here
+    tweetUrl:
+      'https://twitter.com/' + tweet.user_name ?? '' + '/status/' + tweet.conversation_id ?? '',
+    text: tweet.text ?? '',
+    retweetCount: tweet.retweets ?? 0,
+    likeCount: 0, // Unfortunately this API does not deliver like counts, but we will need this count for the other issues
```

Comment:
If like count is zero, it doesn’t mean that there isn’t any like count. Maybe the information is not exist there. in that case, that means you give wrong information to user.